src/DweetCard.tsx

```diff
@@ -260,8 +283,16 @@ export const DweetCard: React.FC<Props> = (props) => {
           Fullscreen
         </a>
       </div>
-      <div className="mb-3">
-        <UserView user={dweet.author} />
+      <div style={{
+        display:"inline-block",
+          width: "100%",
+      }}>
+        <div className="mb-3" style={{
+          float:"left"
+        }}>
+          <UserView user={dweet.author} />
+        </div>
+        { RemixOf(dweet.remix_of) }
```

Comment:
We usually encourage our developers to use a component instead of calling a function:

```tsx
<RemixOf dweet={dweet.remix_of} />

// and

<Avatar user={props.user} link />
```

I'm not sure of all the pros and cons, but a couple of pros are:

- It is clear that a React Element will be returned here
- We could enhance performance of that component later, using [`React.memo()`](https://linguinecode.com/post/prevent-re-renders-react-functional-components-react-memo).
  I'm not sure if we can do to a plain function.
- The arguments being passed are self-describing

Cons:
- The arguments being passed are verbose!
- If we wanted to have this return a plain string in future, rather than a React Element, then that is possible with a function.

Reply:
plain functions are probably faster, but then RemixOf is also not actually being used as a react component. The code is not converted to React.createElement like it would be with the `<>` syntax, and so the code inside may not use react lifecycles, state, and may not avoid running on every render with React.memo. What happens when using a function is basically that whatever is inside it is part of _this_ component as far as React is concerned.

Reply:
I would not really be worried about performance for something like this, here though, just to be clear. I think you should stick to your policy @joeytwiddle :+1:

Reply:
Good point, it will act as if all the RemixOf code was just inline in the DweetCard.

I dare say I don't know how much breaking up the tree into separate components helps or hurts optimisation!

Reply:
Thanks for the feedback! Using a component seems more in line with common React practice, so I have refactored.

Is it common to end up with tons and tons of files with small components in them in a react project?

Reply:
I think it is!

We keep some of our related components together in one file. (E.g. one file for various buttons, one file with a consistent set of form element, one file for layout components that work together.)

But some teams insist on a separate file for each component. (Then using folders to group related components together.)

Using `export const` (as you have done) makes it easy to switch between styles anyway.